src/metrax/nlp_metrics.py

```diff
@@ -285,6 +299,159 @@ def compute(self) -> jax.Array:
+      order: The order 'N' of n-grams to consider. Must be positive.
+
+    Returns:
+      A RougeN metric instance with accumulated per-instance scores.
+    """
+    total_precision = 0.0
+    total_recall = 0.0
+    total_f1 = 0.0
+    num_examples = 0.0
+
+    for pred_str, ref_str in zip(predictions, references):
+      pred_tokens = pred_str.split()
+      ref_tokens = ref_str.split()
+
+      pred_ngrams_counts = _get_single_n_grams(pred_tokens, order)
+      ref_ngrams_counts = _get_single_n_grams(ref_tokens, order)
+      overlap_counts = pred_ngrams_counts & ref_ngrams_counts
+
+      prediction_ngrams = jnp.array(sum(pred_ngrams_counts.values()))
+      reference_ngrams = jnp.array(sum(ref_ngrams_counts.values()))
+      overlapping_ngrams = jnp.array(sum(overlap_counts.values()))
+
+      precision = base.divide_no_nan(overlapping_ngrams, prediction_ngrams)
+      recall = base.divide_no_nan(overlapping_ngrams, reference_ngrams)
+      f1 = base.divide_no_nan(2 * precision * recall, precision + recall)
+
+      total_precision += precision
+      total_recall += recall
+      total_f1 += f1
+      num_examples += 1
+
+    return cls(
+        order=order,
+        total_precision=jnp.array(total_precision, dtype=jnp.float32),
+        total_recall=jnp.array(total_recall, dtype=jnp.float32),
+        total_f1=jnp.array(total_f1, dtype=jnp.float32),
+        num_examples=jnp.array(num_examples, dtype=jnp.float32),
+    )
+
+  def merge(self, other: 'RougeN') -> 'RougeN':
+    """Merges this RougeN metric with another.
+
+    Args:
+      other: Another RougeN metric instance.
+
+    Returns:
+      A new RougeN metric instance with combined statistics.
+    """
+    if self.order != other.order:
+      raise ValueError(
+          'RougeN metrics with different orders cannot be merged. '
+          f'Got {self.order} and {other.order}.'
+      )
+    return RougeN(
+        order=self.order,
+        total_precision=(self.total_precision + other.total_precision),
+        total_recall=(self.total_recall + other.total_recall),
+        total_f1=(self.total_f1 + other.total_f1),
+        num_examples=(self.num_examples + other.num_examples),
+    )
+
+  def compute(self) -> jax.Array:
+    """Computes macro-averaged ROUGE-N recall, precision, and F1-score.
+
+    Returns:
+      A JAX array where:
+            - index 0: macro-averaged precision
+            - index 1: macro-averaged recall
+            - index 2: macro-averaged f1score (derived from avg_precision and
+            avg_recall)
+          Scores are 0.0 if num_examples is zero.
+    """
+    macro_avg_precision = base.divide_no_nan(
+        self.total_precision, self.num_examples
+    )
+    macro_avg_recall = base.divide_no_nan(self.total_recall, self.num_examples)
+    macro_avg_f1score = base.divide_no_nan(self.total_f1, self.num_examples)
+    return jnp.stack([macro_avg_precision, macro_avg_recall, macro_avg_f1score])
```

Comment:
Is it common for users to want all three of these as a bundle for the most part? (as opposed to separate precision/recall/f1 metrics)

Reply:
I meant more would it make sense to just have 3 separate metrics for these, or is it common enough to want all of them that it makes sense to bundle them together? There is likely a reason Keras did it that way but would prefer not to copy over design decisions without understanding first

Reply:
Looks like the decision originated here:
https://github.com/keras-team/keras-hub/pull/122#discussion_r882341574

In an ideal world I think it would be cleaner if these were separate metric classes. But realistically SG to move forward with this way and make a bug to revisit the decision.